chess.js

```diff
@@ -27,19 +27,50 @@ function isCheck() {
 }
 
 function isValidKingMove(from, to) {
-    return false
+    const positionFrom = getPosition(from)
+    const positionTo = getPosition(to)
+    const rowDifference = Math.abs(positionFrom.row - positionTo.row)
+    const columnDifference = Math.abs(positionFrom.column - positionTo.column)
+    return (
+        (rowDifference === 0 || rowDifference === 1) && (columnDifference === 0 || columnDifference === 1)
+    )
 }
 function isValidQueenMove(from, to) {
-    return true
+    const positionFrom = getPosition(from)
+    const positionTo = getPosition(to)
+    const rockMove = positionFrom.row === positionTo.row || positionFrom.column === positionTo.column;
+    const bishopMove = Math.abs(positionFrom.row - positionTo.row ) === Math.abs(positionFrom.column - positionTo.column)
+    return (
+        rockMove || bishopMove
+    )
 }
```

Comment:
Ovo može mnogo kraće, probaj da pogodiš kako (blizu si već).

Reply:
function isValidQueenMove(from, to) {
    const queenRookMove = isValidRookMove(from,to)
    const queenBishopMove = isValidBishopMove(from,to)
    return queenBishopMove || queenRookMove
}

???

Reply:
Bravo :) Samo što možeš inline obe te promenljive pa da bude sve u jednom redu.